### DISCUSSION TITLE
v1.63.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#2927] [Announcements]
# v1.63.0

Hello, Immich fans! Welcome to the `v1.63.0` release of Immich. There are many updates in this version that we hope you'll like. Some of the key highlights include:

- Initial support for a read-only/existing library.
- Support for more raw formats.
- Album titles now appear in the search results.
- Pinch-to-zoom asset grid on the mobile app.
- Facial recognition on the mobile app.

### Initial support for read-only/existing library.

Thanks to @alex-phillips, Immich now has the ability to import an existing gallery without the need to upload the files. This has been one of the most requested features, and I am happy that we can now cover this use case as well. However, this is only the initial implementation and will receive improvements in the future.

The current limitations of this feature are:

- Assets are not automatically synced and must instead be manually synced with the CLI tool.
- Only new files that are added to the gallery will be detected.
- Deleted and moved files will not be detected.

**To understand how to use this feature, please see head to the [documentation site](https://immich.app/docs/features/read-only-gallery)**

### Support for more raw formats

With the integration of `imagemagick`, Immich now has support more raw photo formats and will generate high quality thumbnails to display on the timeline. I hope this helps everyone who owns a DSLR or mirrorless camera.

### More love to the mobile app

We have neglected the mobile app for a while, so we’ve decided to give it some left over love from the Web and the Server 😛. 

We added the `facial recognition` feature to the mobile app so you can view faces and photos of a person. 

<p align="center">
<img src="https://github.com/immich-app/immich/assets/27055614/d23268c1-40b2-42b8-aed0-148ce58c43f7" width="250"> 

<img src="https://github.com/immich-app/immich/assets/27055614/58da6c5a-925e-471f-aa7e-0fe423c3b0fc" width="250"> 

<img src="https://github.com/immich-app/immich/assets/27055614/0f516ad0-9cc4-41a5-b5da-7148a511ee05" width="250"> 

</p>

You can pinch to zoom in and out of the timeline to change the number of assets displayed on each row.

[pinch2zoom.webm](https://github.com/immich-app/immich/assets/10599762/ec6bc6b3-10d7-4c43-8119-861719044282)

- - - -

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="Loading ~4000 images/videos"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

🎉 Cheers! 🎉

### Web
* feat(web): add zoom toggle icon by @mtaberna in https://github.com/immich-app/immich/pull/2873
* feat(web): Add album sorting to albums view by @klejejs in https://github.com/immich-app/immich/pull/2861
* feat(web): custom drop down button by @alextran1502 in https://github.com/immich-app/immich/pull/2887
* feat(web): add album to search result by @alextran1502 in https://github.com/immich-app/immich/pull/2900
* feat(web): Add keyboard event support to memory view by @klejejs in https://github.com/immich-app/immich/pull/2890
* fix(web): redirect to parent folder if asset viewer was open before reloading by @Sophtli in https://github.com/immich-app/immich/pull/2863
* fix(web): Mov files should show up in file picker. by @samip5 in https://github.com/immich-app/immich/pull/2886
* fix(web): keep video volume by @faupau03 in https://github.com/immich-app/immich/pull/2897
* fix(web): memory pause autoplay when scrolling down by @alextran1502 in https://github.com/immich-app/immich/pull/2923

### Server
* feat(server): Add support for many missing raw formats by @elliotclee in https://github.com/immich-app/immich/pull/2834
* feat(server): JPEG XL by @uhthomas in https://github.com/immich-app/immich/pull/2893
* feat(server): M2TS by @uhthomas in https://github.com/immich-app/immich/pull/2896
* feat(server): support for read-only assets  and importing existing items in the filesystem by @alex-phillips in https://github.com/immich-app/immich/pull/2715
* fix(server): use HTTP status OK instead of CREATED for assets by @uhthomas in https://github.com/immich-app/immich/pull/2876
* fix(server): Fix person's assets retrieval order by @alextran1502 in https://github.com/immich-app/immich/pull/2920
* fix(server): transform isReadOnly DTO to boolean by @alextran1502 in https://github.com/immich-app/immich/pull/2912
* chore(server): remove openapi assertion for dart by @alextran1502 in https://github.com/immich-app/immich/pull/2916
* chore(server): Improve moveAsset log by @alextran1502 in https://github.com/immich-app/immich/pull/2878
* chore(server): patch dart openapi assertion by @alextran1502 in https://github.com/immich-app/immich/pull/2914
* chore(server): add Canon CR3 mime type by @416c616e in https://github.com/immich-app/immich/pull/2922
* refactor(server, web): create shared link by @jrasm91 in https://github.com/immich-app/immich/pull/2879

### Mobile
* feat(mobile): Adds photo thumbnail to videos by @martyfuhry in https://github.com/immich-app/immich/pull/2880
* fix(mobile): Fixes local position late initialization by @martyfuhry in https://github.com/immich-app/immich/pull/2884
* feat(mobile): pinch to zoom on asset grid by @fyfrey in https://github.com/immich-app/immich/pull/2905
* feat(mobile): optimize screen space usage by @fyfrey in https://github.com/immich-app/immich/pull/2911
* feat(mobile): Facial recognition by @alextran1502 in https://github.com/immich-app/immich/pull/2507

### Chore
* fix(deployment) Add dependency on immich-web to immich-proxy by @etnoy in https://github.com/immich-app/immich/pull/2875
* chore(doc): move public msg to general section by @stavros-k in https://github.com/immich-app/immich/pull/2864
* chore(readme): mention offline support feature by @fyfrey in https://github.com/immich-app/immich/pull/2902

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.63.0'>v1.63.0</a>.</em>

**Comment by tbleiker** [comment]
I like the new "read-only/existing library" feature. Is it possible to add the same library/folder to multiple users? What happens if I set the same path for multiple users and run the cli for each of them?

And once more: thx a lot to all who put time and passion into this great project! 

Edit: typo

**Reply by etnoy** [reply]
It will probably work but it's not tested. Feel free to report the results!

**Reply by alex-phillips** [reply]
Currently there is a unique check on the file path. You can set the same path for multiple users, but only one user can import each file. If another user attempts to the API will respond that the file is a duplicate. I believe the best way to handle multiple users accessing the same imported files is to share them

**Comment by 0xEmma** [comment]
Was v1.63.1 skipped with the v1.63.2 release?

**Reply by alextran1502** [reply]
Yes, I triggered the release pipeline this morning but forgot actually to release it. So after additional fixes, they are all in the new 1.63.2 release

**Comment by charles-997** [comment]
@alex-phillips thank you for your contribution around the existing folder support.  Do you have a github sponsor or something setup for users to send some thanks your way?

**Reply by alex-phillips** [reply]
@charles-997 I don't personally, but I would encourage and thanks for donations to benefit the project directly! There are several links at the bottom of the project README you can use.

**Reply by charles-997** [reply]
> @charles-997 I don't personally, but I would encourage and thanks for donations to benefit the project directly! There are several links at the bottom of the project README you can use.

Agreed!  Already done, I am now a GitHub sponsor of the immich project. Thank you for your contributions.
